### PR TITLE
Build new releases on merge to master

### PR DESCRIPTION
**Pull request by whereswaldon**
This pull request *should* configure TravisCI building and publishing a release each time we update the master branch. I have tested it as well as I know how to in an isolated environment. Unfortunately, the only way to see that works *for sure* is to merge it.

### CODE REVIEW

**Comment by whereswaldon** [comment]
Had to do some updates after posting, but I believe that it will work when we hit `merge`.

**Comment by whereswaldon** [comment]
Good suggestion. I'm doing that now

**Comment by whereswaldon** [comment]
Okay, that testing did reveal a problem in the Travis environment that I didn't anticipate. Thanks @Caton101. [Here's](https://travis-ci.org/whereswaldon/muscadine/builds/480152372) an example of a Travis build that built a [release](https://github.com/whereswaldon/muscadine/releases/tag/release-3cdc096) using this code.

File: build-releases.sh [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+set -euo pipefail
+
+# only build on master
+if [ "$TRAVIS_PULL_REQUEST" != "false" ] || [ "$TRAVIS_BRANCH" != "master" ]; then
+    echo "Not building release, this is a pull request"
+    exit 0
+fi
+
+# get hub if we don't have it
```
Comment (jwhett) [inline build-releases.sh]: `get hub` <-- this hurt my brain to read

File: build-releases.sh [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    go get github.com/github/hub
+fi
+
+release_flags=""
+head_commit="$(git rev-parse HEAD 2> /dev/null)"
+
+# build pre-release if not on a tag, otherwise build release
+if ! git describe --tags --exact-match HEAD > /dev/null 2>&1 ; then
+    echo "Building pre-release, not tagged commit"
+    readonly tag="release-$(echo ${head_commit} | head -c 7)"
```
Comment (jwhett) [inline build-releases.sh]: thank you for using `readonly`

File: build-releases.sh [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+readonly bin_name="$project"
+readonly message="Automatic build"
+
+# create the release and upload artifacts
+hub release create $release_flags --message="$message" --commitish="$head_commit" "$tag"
+for os in darwin linux windows openbsd; do
+    archive_name="$project-$os.tar.gz"
+    echo "Building $project for $os"
+    env GOOS="$os" CGO_ENABLED=0 go build -o "$bin_name" &&\
+    tar czf "$archive_name" "$bin_name" &&\
```
Comment (jwhett) [inline build-releases.sh]: this does mean that you have to install more third-party tools to unpack windows release files. i'm not going to block on this as there are several well-known tools for this and the target audience is sure to know what they are and how to find/install them.

**Comment by Caton101** [comment]
Making Windows users download 7zip or another archive utility is very funny to me.

**Comment by jwhett** [comment]
@Caton101, I resemble that remark! 😆

**Comment by Caton101** [comment]
Should we delete the branch to prevent stale branches?

**Pull request merged**